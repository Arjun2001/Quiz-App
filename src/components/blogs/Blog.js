import React, { useState, useEffect } from 'react'
import axios from 'axios'
import { makeStyles } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper';
import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';
import "./blog.css";

const useRowStyles = makeStyles({
    root: {
      '& > *': {
        borderBottom: 'unset',
      },
    },
  });


function createData(name, time,code,start,end) {
return {
    name,
    time,
    history: [{ code: code, start: start, end: end }]
};
}

function Row(props) {
    const { row } = props;
    const [open, setOpen] = React.useState(false);
    const classes = useRowStyles();
  
    return (
      <React.Fragment>
        <TableRow className={classes.root}>
          <TableCell>
            <IconButton aria-label="expand row" size="small" onClick={() => setOpen(!open)}>
              {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
            </IconButton>
          </TableCell>
          <TableCell component="th" scope="row">
            {row.name}
          </TableCell>
          <TableCell align="right">{row.time}</TableCell>
        </TableRow>
        <TableRow>
          <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={6}>
            <Collapse in={open} timeout="auto" unmountOnExit>
              <Box margin={1}>
                <Typography variant="h6" gutterBottom component="div">
                  Details
                </Typography>
                <Table size="small" aria-label="purchases">
                  <TableHead>
                    <TableRow>
                      <TableCell>Subject</TableCell>
                      <TableCell>Start Time</TableCell>
                      <TableCell align="right">End Time</TableCell>
                    </TableRow>
                  </TableHead>
                  <TableBody>
                    {row.history.map((historyRow) => (
                      <TableRow key={historyRow.code}>
                        <TableCell component="th" scope="row">
                          {historyRow.code}
                        </TableCell>
                        <TableCell>{historyRow.end}</TableCell>
                        <TableCell align="right">{historyRow.end}</TableCell>
                      </TableRow>
                    ))}
                  </TableBody>
                </Table>
              </Box>
            </Collapse>
          </TableCell>
        </TableRow>
      </React.Fragment>
    );
  }

const Blog = (props) =>{

    const [rows, setRows] = useState([]);
    const [len,setLen] = useState(0);


    const getData = async () => {
        try {
            await axios ({
                method:'post',
                url: "http://35.225.238.45:5000/api/contests",
                headers: {
                    "Authorization":`Bearer ${localStorage.getItem('Token')}`,
                    "Content-Type": "application/json"
                },
                data: {
                    code:"None"
                }
            }).then(res => {
                let rows1 = []
                for(var i =0;i <res.data.length;i++) {
                    let now = new Date();
                    now = now.toISOString();
                    if ((now < res.data[i].START)) {
                    rows1.push(createData(res.data[i].NAME,duration(res.data[i].START),res.data[i].CODE,convertTime(res.data[i].START),convertTime(res.data[i].END)))
                    }
                }
            setRows(rows1)
            setLen(rows1.length)
            })
        }catch (err) {
            console.log(err);
        }
    }
    
    useEffect(() => {
        getData();
    }, [])

    const convertTime = (START) => {
        let time = new  Date(START);
        time = time.toLocaleString("en-SG");
        return time;
    }
    const duration = (a,) => {
        let time1 = new  Date(a);
        let d = (Math.floor(Math.abs(new Date() - time1)/(1000 * 3600 * 24)))
        console.log(d)
        if (d === 0) {
            d = "Today"
        } else {
            d = d + " days"
        }
        console.log(d)
        return d
    }

    return(
        <div className="blogs-container">
            <div className="blog-title">Latest Updates</div>
            <div className="separator"><span style={{opacity:"0.34"}}>{len} item</span></div>
            <TableContainer component={Paper}>
            <Table aria-label="collapsible table">
                <TableHead>
                <TableRow>
                    <TableCell />
                    <TableCell>Contest</TableCell>
                    <TableCell align="right">In</TableCell>
                </TableRow>
                </TableHead>
                <TableBody>
                {rows.map((row) => (
                <Row key={row.code} row={row} />
                ))}
                </TableBody>
            </Table>
            </TableContainer>
        </div>
    )
}

export default Blog;