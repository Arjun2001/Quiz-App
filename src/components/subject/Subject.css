.sub-main {
    padding: 10px;
    text-align: center;
    background: white;
    border-radius: 10px;
    width: 100%;
    border: 5px solid #b67162;
}

.cards {
    display: flex;
    flex-direction: row;
    flex-wrap:wrap;
}

.MuiPaper-root {
    border-radius: 15px !important;
    margin: 0px 15px 15px 0px;
}

.click-class {
    cursor: pointer;
}

.makeStyles-paper-13{
    width: 550px;
}

.MuiCardContent-root {
    padding-bottom: 5px !important;
}