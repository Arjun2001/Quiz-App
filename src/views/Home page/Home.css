.home-main {
    display: flex;
    flex-direction: row;
    padding: 25px;
    height: 100vh;
}

.left-main {
    display: flex;
    flex: 0.7;
    margin-right: 10px;
}

.right-main {
    display: flex;
    flex: 0.3;
}