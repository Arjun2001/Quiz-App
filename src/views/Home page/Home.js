import React from 'react'
import Navbar from '../../components/Navbar/Navbar'
import Blog from '../../components/blogs/Blog'

function Home() {
    return (
        <div>
            <Navbar />
            <div>
                Hello , you are on home page!
                {/* <Blog></Blog> */}
            </div>
            
        </div>
    )
}

export default Home
