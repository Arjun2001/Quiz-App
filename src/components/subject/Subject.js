import React, {useState} from 'react'
import './Subject.css'
import SubCard from './SubCard'
import AddCard from './AddCard'

import { makeStyles } from '@material-ui/core/styles';
import Modal from '@material-ui/core/Modal';
import Backdrop from '@material-ui/core/Backdrop';
import Fade from '@material-ui/core/Fade';
import { Button, Form, FormGroup, Label, Input, FormText } from 'reactstrap';

function Subject() {

    const [open, setOpen] = useState(false);

    const useStyles = makeStyles((theme) => ({
        modal: {
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
        },
        paper: {
          backgroundColor: theme.palette.background.paper,
          border: '2px solid #000',
          boxShadow: theme.shadows[5],
          padding: theme.spacing(2, 4, 3),
        },
      }));
      const classes = useStyles();
    const handleOpen = () => {
    setOpen(true);
    };

    const handleClose = () => {
    setOpen(false);
    };
    
    return (
        <div className='sub-main'>
            <h3>SUBJECTS</h3>
            <hr></hr>
            <div className="cards">
                <SubCard />
                <AddCard open={handleOpen}/> 
            </div>
            <Modal
                aria-labelledby="transition-modal-title"
                aria-describedby="transition-modal-description"
                className={classes.modal}
                open={open}
                onClose={handleClose}
                closeAfterTransition
                BackdropComponent={Backdrop}
                BackdropProps={{
                timeout: 500,
                }}
            >
                <Fade in={open}>
                <div className={classes.paper}>
                <Form>
                    <FormGroup>
                        <Label for="exampleEmail">Course Code</Label>
                        <Input type="text" name="course_code" id="exampleEmail" />
                    </FormGroup>
                    <FormGroup>
                        <Label for="exampleEmail">Course Name</Label>
                        <Input type="text" name="course_name" id="exampleEmail" />
                    </FormGroup>
                    <FormGroup>
                        <Label for="exampleEmail">Course Photo</Label>
                        <Input type="text" name="course_name" id="exampleEmail"/>
                    </FormGroup>
                    <FormGroup>
                        <Label for="exampleEmail">Course Description</Label>
                        <Input type="text" name="course_name" id="exampleEmail" />
                    </FormGroup>
                    <Button>Submit</Button>
                </Form>
                </div>
                </Fade>
            </Modal>
        </div>
    )
}

export default Subject
