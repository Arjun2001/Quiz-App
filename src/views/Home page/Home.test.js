import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Home from './Home';
import Blog from '../../components/blogs/Blog'
import Subject from '../../components/subject/Subject'

describe('Blog component', () => {
    it('div rendering', () => {
        const wrapper = shallow(<Home />);
        wrapper.find('div.home-main');
        expect(wrapper.containsMatchingElement(<Blog />)).toEqual(true);
        expect(wrapper.containsMatchingElement(<Subject />)).toEqual(true);
    });
    it('class rendering', () => {
        const wrapper = shallow(<Home />);
        wrapper.find('div.left-main');
        wrapper.find('div.right-main');
    });
    it('matches the snapshot', () => {
        const tree = renderer.create(<Home />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});