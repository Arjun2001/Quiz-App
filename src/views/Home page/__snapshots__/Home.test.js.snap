// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog component matches the snapshot 1`] = `
<div>
  <nav
    className="navbar-bg navbar navbar-expand navbar-dark"
  >
    <a
      className="navbar-brand"
      href="/home"
    >
      <img
        alt="quiz-logo"
        src="https://cdn.pixabay.com/photo/2018/08/31/11/17/quiz-3644414_960_720.png"
        style={
          Object {
            "width": "60px",
          }
        }
      />
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      />
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="btn btn-outline-warning"
          href="/start"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "marginRight": "15px",
            }
          }
        >
          Demo
        </a>
        <span
          className="navbar-text"
        >
          Signed in as: 
          <strong
            style={
              Object {
                "color": "white",
              }
            }
          />
        </span>
        <div
          className="nav-item dropdown"
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            data-rb-event-key={null}
            href="#"
            id="basic-nav-dropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
          />
        </div>
      </div>
    </div>
  </nav>
  <div
    className="home-main"
  >
    <div
      className="left-main"
    >
      <div
        className="sub-main"
      >
        <h3>
          SUBJECTS
        </h3>
        <hr />
        <div
          className="cards"
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexWrap": "wrap",
              }
            }
          >
            <div />
          </div>
        </div>
      </div>
    </div>
    <div
      className="right-main"
    >
      <div
        className="blogs-container"
      >
        <div
          className="blog-title"
        >
          Latest Updates
        </div>
        <div
          className="separator"
        >
          <span
            style={
              Object {
                "opacity": "0.34",
              }
            }
          >
            0
             item
          </span>
        </div>
        <div
          className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <table
            aria-label="collapsible table"
            className="MuiTable-root"
            role={null}
          >
            <thead
              className="MuiTableHead-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                />
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Contest
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                  scope="col"
                >
                  In
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root"
              role={null}
            />
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
